### PR TITLE
DOC, MAINT: Prepare for 1.16.0rc1 release.

### PR DESCRIPTION
**Pull request by charris**
- Create 1.16.0-changelog.rst
- Prepare 1.16.0-notes.rst
- Update tools/changelog.py for Python 3.

[ci skip]

### CODE REVIEW

**Comment by charris** [comment]
The release notes need significant work, but I put this up early as a spot to work on them. Any help would be appreciated.

**Comment by tylerjereddy** [comment]
I'm taking a look now

**Comment by tylerjereddy** [comment]
For "improvements" subsection maybe:
- Chain exceptions to give better error messages for invalid PEP3118 format strings
- Einsum optimization path updates / efficiency improvements
- Allow use of svd on empty arrays
- `np.angle` now works on subclasses of `ndarray` like `ma.array`
- `np.expand_dims` now works on subclasses of `ndarray`
- choosing (forcing) `NPY_1_7_API_VERSION` or earlier no longer emits a compiler warning

**Comment by rgommers** [comment]
What's in here LGTM, didn't check for omissions.

File: tools/changelog.py [hunk 1]
```diff
@@ -111,7 +113,7 @@ def main(token, revision_range):
     # document authors
     authors = get_authors(revision_range)
     heading = u"Contributors"
-    print()
+    print(u"")
```
Comment (eric-wieser) [inline tools/changelog.py]: Does this behave differently?

**Comment by mattip** [comment]
> deprecate passing a generator to stack functions (has already affected SciPy and scikit-learn)

This does not issue a warning
```
>>> np.stack((range(10), [x for x in range(10)]))
array([[0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
       [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]])
```

**Comment by mattip** [comment]
I pushed some changes, adding a mention of CI additions and improvements and some rewording.

**Comment by eric-wieser** [comment]
> This does not issue a warning

That's not passing a generator at all, that's passing a `Tuple[range, list]`.

Even `Tuple[Generator[...], ...]` is still supported.

We should clarify that message though - it's not generators that are deprecated, its itera**tors**.

File: tools/changelog.py (hunk 1 above)
Comment (charris) [reply tools/changelog.py]: I thought it did, but it looks like the problem was an artifact of the UTF8Writer.

**Comment by charris** [comment]
> It is under changes at the moment

It' quite alright to have thing in two places.

**Comment by charris** [comment]
Updated.

**Comment by charris** [comment]
Good enough for the rc.

**Pull request merged**